test/support/utils.h

```diff
@@ -754,6 +755,18 @@ struct MaxAbsFunctor<::std::complex<_Tp>>
     }
 };
 
+struct tuple_add_fn
+{
+    template <typename Tup1, typename Tup2>
+    auto operator()(Tup1 lhs, const Tup2& rhs) const
+    {
+        using ::std::get;
+        get<0>(lhs) += get<1>(rhs);
+        get<1>(lhs) += get<1>(rhs);
+        return lhs;
+    }
```

Comment:
I don't have a strong preference here, but I'm curious why not use `const Tup1& lhs` and make a new tuple for the return?
Also, for the way we use this functor, can't we just have a single template argument for `lhs`, `rhs` and return type?

Reply:
I have switched to taken both parameters by const reference and creating a new tuple to return.

The reason we can't use a single template argument is due to constness differences between `lhs` and `rhs`. `lhs` will always have decayed tuple elements since it is the type of the tuple that collects the reduction value. The type of `rhs` may differ since it is passed directly from the input values buffer [here](https://github.com/oneapi-src/oneDPL/blob/a678aee8b1afd1b4fe098897e0cf88a0743e7386/include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h#L945C67-L945C67). Using a single template type causes a compiler error.